File: benchmark/php/task5_class.php
<?php
class Point {
    public function __construct(
        public float $x,
        public float $y
    ) {}

    public function distanceTo(Point $other): float {
        $dx = $this->x - $other->x;
        $dy = $this->y - $other->y;
        return sqrt($dx * $dx + $dy * $dy);
    }
}

File: benchmark/r/task5_class.r
Point <- setRefClass("Point",
  fields = list(x = "numeric", y = "numeric"),
  methods = list(
    initialize = function(x = 0, y = 0) {
      .self$x <- x
      .self$y <- y
    },
    distance_to = function(other) {
      sqrt((x - other$x)^2 + (y - other$y)^2)
    }
  )
)
